server/server.go

```diff
@@ -531,6 +531,7 @@ func (s *Server) onConn(conn *clientConn) {
 	logutil.Logger(ctx).Debug("new connection", zap.String("remoteAddr", conn.bufReadConn.RemoteAddr().String()))
 
 	defer func() {
+		terror.Log(conn.Close())
```

Comment:
`conn.Close()` should have been called in `conn.Run()` function, line 559...
It's better to figure out why it's not called there.

Reply:
I think the reason it can't be called there, is the plugin hooks that follow expect the connection is still established. So it would have to come after.

But I looked far back in the code, and this bug has always existed since before plugins. It is less obvious though with optimistic locking...

Reply:
If so, how about removing the `conn.Close` from `conn.Run()` function?
Keep the code in two places would reduce maintainability

Reply:
Done